Doc/reference/toplevel_components.rst

```diff
@@ -85,7 +85,7 @@ Interactive input
 
 Input in interactive mode is parsed using the following grammar:
 
-.. productionlist::
+.. productionlist:: interactive-input-component
    interactive_input: [`stmt_list`] NEWLINE | `compound_stmt` NEWLINE
```

Comment:
This change breaks the link to stmt_list.

Should we use the same identifier here, in expressions, simple statements and compound statements?

Reply:
Yeah, I think that is a good idea (to preserve those links). Is there any other missing links between sections?

Reply:
Or maybe use the least amount of namespaces possible?

Reply:
I didn't check all modified pages.